ui/pages/confirmations/hooks/alerts/signatures/useDomainMismatchAlerts.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest,
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+export default function useDomainMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const { msgParams } = currentConfirmation || {};
+
+  const isSIWE = isSIWESignatureRequest(currentConfirmation);
+  const isInvalidSIWEDomain =
+    isSIWE && !isValidSIWEOrigin(msgParams as WrappedSIWERequest);
```

Comment:
It will be useful to include above 3 lines also inside `useMemo`

Reply:
I had this originally and then @matthewwalsh0 brought up a good point to reduce the # of observed dependencies